src/modes/world.cpp

```diff
@@ -1610,3 +1569,129 @@ void World::setAITeam()
     Log::debug("World", "Blue AI: %d red AI: %d", m_blue_ai, m_red_ai);
 
 }   // setAITeam
+
+// As a class name can't be skipped with "using", we use a preprocessor macro
+// to clean up the two following functions
+#define ACS AchievementsStatus
+
+//-----------------------------------------------------------------------------
+/* This function takes care to update all relevant achievements
+ * and statistics counters related to a finished race. */
+void World::updateAchievementDataEndRace()
+{
+    const unsigned int kart_amount = getNumKarts();
+
+    for(unsigned int i = 0; i < kart_amount; i++)
+    {
+        // TODO : does this work in multiplayer ?
```

Comment:
Should be checked, and the TODO removed

Reply:
I'm almost certain it doesn't work, but I'm nearly as sure that it doesn't work either in current master (the core logic in calling the PlayerManager hasn't really changed).

I'll do some tests and post the results back.